typings/index.d.ts

```diff
@@ -14,6 +14,10 @@ declare namespace commander {
   }
   type CommanderErrorConstructor = new (exitCode: number, code: string, message: string) => CommanderError;
 
+  interface InvalidOptionArgumentError extends CommanderError {
+  }
+  type InvalidOptionArgumentErrorConstructor = new (message: string) => InvalidOptionArgumentError;
```

Comment:
Question: why in commander typings there are separations of class types into interfaces and constructor types?

So here instead of:
```ts
class InvalidOptionArgumentError extends CommanderError {
  constructor(message: string);
}
```
There is:
```ts
interface InvalidOptionArgumentError extends CommanderError {
}
type InvalidOptionArgumentErrorConstructor = new (message: string) => InvalidOptionArgumentError;
```

(To get a constructor type from a class variant - `typeof InvalidOptionArgumentError`.)

Reply:
The pattern is to allow export of both a global object and some types. See #1081

Reply:
So, i cloned [fson/commander-ts-repro](https://github.com/fson/commander-ts-repro) and could not reproduce the issue. Then i changed typescript dependency there from `^3.6.3` to simply `3.6.3` and i saw the expected error. With `^3.6.3` typescript was resolving to `3.9.7`.

Is it possible that the issue was fixed on typescript end and this design is no longer needed?

Reply:
Yes it is possible has been fixed or improved in TypeScript, I have not checked again since the original problem! I am planning to rework the exports and typings when ESM support is out of prototype in node.